godtools/App/Features/Menu/Domain/UseCases/GetUserActivityBadgeUseCase/GetUserActivityBadgeUseCase.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  GetUserActivityBadgeUseCase.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 1/24/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import GodToolsToolParser
+import SwiftUI
+
+class GetUserActivityBadgeUseCase {
+    
+    private let localizationServices: LocalizationServices
+    
+    init(localizationServices: LocalizationServices) {
+        
+        self.localizationServices = localizationServices
+    }
+    
+    func getBadge(from godToolsSharedLibraryBadge: Badge) -> UserActivityBadgeDomainModel {
+        
+        let badgeType = godToolsSharedLibraryBadge.type
+        let variant = Int(godToolsSharedLibraryBadge.variant)
+        
+        let badgeText = getBadgeText(badgeType: badgeType, progressTarget: Int(godToolsSharedLibraryBadge.progressTarget))
+        let iconName = getIconImageName(badgeType: badgeType, variant: variant)
+        
+        let iconBackgroundColor = Color(godToolsSharedLibraryBadge.colors.containerColor(mode: .light))
+        let iconForegroundColor = Color(godToolsSharedLibraryBadge.colors.color(mode: .light))
+        
+        return UserActivityBadgeDomainModel(
+            badgeText: badgeText,
+            badgeType: badgeType,
+            iconBackgroundColor: iconBackgroundColor,
+            iconForegroundColor: iconForegroundColor,
+            iconImageName: iconName,
+            isEarned: godToolsSharedLibraryBadge.isEarned,
+            variant: variant
+        )
+    }
+    
+    private func getBadgeText(badgeType: Badge.BadgeType, progressTarget: Int) -> String {
+        
+        if progressTarget == 1 {
+            
+            return getSingularString(for: badgeType)
+            
+        } else {
+            
+            return getPluralString(for: badgeType, progressTarget: progressTarget)
+        }
```

Comment:
does iOS have a native way to handle plurals?

The 1/other rule applies to english, but it's not the same for all foreign languages.

Reply:
a quick google search found this: https://developer.apple.com/documentation/xcode/localizing-strings-that-contain-plurals

Reply:
I'm also not sure how well OneSky supports iOS plurals, I know it has some outdated rules for French plurals on Android

Reply:
For reference, here is where a couple plurals are defined in Android:

https://github.com/CruGlobal/godtools-android/blob/develop/app/src/main/res/values/strings_tool_details.xml#L9-L19

and you can see how those same plurals get translated to Russian:
https://github.com/CruGlobal/godtools-android/blob/develop/app/src/main/res/values-ru/strings_tool_details.xml#L9-L21

Reply:
Ohhhh interesting!  Good catch!  I'm thinking back to other apps I've worked on and wondering now if they were localizing correctly for plurals or not 🤔  Will make a separate ticket.

Reply:
ticket here: https://jira.cru.org/browse/GT-1918